include/openrave/environment.h

```diff
@@ -24,19 +24,21 @@
 
 namespace OpenRAVE {
 
+#include <openrave/config.h>
+
 #if OPENRAVE_ENVIRONMENT_RECURSIVE_LOCK
-#if __cplusplus >= 201703L
+#if OPENRAVE_STD_SCOPED_LOCK
 #include <mutex>
 using EnvironmentMutex = ::std::recursive_mutex;
-using EnvironmentLock  = ::std::unique_lock<std::recursive_mutex>;
+using EnvironmentLock  = ::std::scoped_lock<std::recursive_mutex>;
```

Comment:
this line must be a typo from https://github.com/rdiankov/openrave/pull/1120 , but no one noticed - gcc12 build works as-is.